packages/twenty-server/src/engine/metadata-modules/agent/agent-execution.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable } from '@nestjs/common';
+
+import { anthropic } from '@ai-sdk/anthropic';
+import { openai } from '@ai-sdk/openai';
+import { generateObject } from 'ai';
+import { z } from 'zod';
+
+import { resolveInput } from 'src/modules/workflow/workflow-executor/utils/variable-resolver.util';
+
+import { AgentEntity } from './agent.entity';
+import { AgentException, AgentExceptionCode } from './agent.exception';
+
+import { inferZodSchemaFromExampleResponse } from './utils/infer-zod-schema-from-example-response.util';
+
+@Injectable()
+export class AgentExecutionService {
+  constructor() {}
+
+  private getModelConfig(model: string): {
+    provider: 'openai' | 'anthropic';
+    modelId: string;
+  } {
+    if (model.startsWith('gpt-')) {
```

Comment:
Not a big fan of this, I think we should have clean constants to store things like that